### DISCUSSION TITLE
Image is not being compressed It produces the image with same size

### DISCUSSION THREAD

**Discussion by shuaibaslam2019** [spatie/image-optimizer#180] [Q&A]
The following code does not affect the image size at all.

`
$optimizerChain = (new OptimizerChain)
            ->addOptimizer(new Jpegoptim(['-m85', '--strip-all', '--all-progressive']))
            ->addOptimizer(new Pngquant()); `
`
$optimizerChain->optimize('../public/admin/images/test.jpg', '../public/admin/images/test_low.jpg');
`

Also, I want to compress the image before adding it to the zip and without saving elsewhere, as I am taking images from S3 bucket.

How could I do that?

Thank you!

**Comment by steve231293** [comment]
Hello there,
I have the same problem with PNG images.
After optimized the file size is the same.

Anyone could help me?

**Comment by varemel** [comment]
I faced the same problem. Solution was to install `jpegoptim`.

I found that in OptimizerChain (https://github.com/spatie/image-optimizer/blob/main/src/OptimizerChain.php) this error is logged `php sh: jpegoptim: command not found`

https://github.com/spatie/image-optimizer/blob/02e2717c31334646f00a0e25cf5edbf41bd20ac3/src/OptimizerChain.php#L81

Or try to check what exact error message your particular optimizer returns.
